packages/core/src/utilities/VoxelValue.ts

```diff
@@ -0,0 +1,235 @@
+import type { BoundsIJK, Point3, VolumeScalarData } from '../types';
+
+/**
+ * This is a simple, standard interface to values associated with a voxel.
+ */
+export default class VoxelValue<T> {
+  public modifiedSlices = new Set<number>();
+  public boundsIJK = [
+    [Infinity, -Infinity],
+    [Infinity, -Infinity],
+    [Infinity, -Infinity],
+  ] as BoundsIJK;
+
+  // Provide direct access to the underlying data, if any
+  public scalarData: VolumeScalarData;
+  public map: Map<number, T>;
+  public sourceVoxelValue: VoxelValue<T>;
+  public isInObject: (pointIPS, pointIJK) => boolean;
+
+  points: Set<number>;
+  dimensions: Point3;
+  width: number;
+  frameSize: number;
+  _get: (index: number) => T;
+  _set: (index: number, v: T) => boolean | void;
+
+  constructor(
+    dimensions,
+    _get: (index: number) => T,
+    _set?: (index: number, v: T) => boolean | void
+  ) {
+    this.dimensions = dimensions;
+    this.width = dimensions[0];
+    this.frameSize = this.width * dimensions[1];
+    this._get = _get;
+    this._set = _set;
+  }
+
+  public getIJK = (i, j, k) => {
+    const index = i + j * this.width + k * this.frameSize;
+    return this._get(index);
+  };
+
+  public setIJK = (i: number, j: number, k: number, v) => {
+    const index = i + j * this.width + k * this.frameSize;
+    if (this._set(index, v) !== false) {
+      this.modifiedSlices.add(k);
+      VoxelValue.addBounds(this.boundsIJK, [i, j, k]);
+    }
+  };
+
+  public addPoint(point: Point3 | number) {
+    const index = Array.isArray(point)
+      ? point[0] + this.width * point[1] + this.frameSize * point[2]
+      : point;
+    if (!this.points) {
+      this.points = new Set<number>();
+    }
+    this.points.add(index);
+  }
```

Comment:
it is not quite evident what is this, since I expected this class to handle images, but seems like it can handle points too, add docs please

Reply:
Maybe all `this.points` stuff should be a separate manager?